packages/ag-charts-community/src/chart/toolbar/toolbar.ts

```diff
@@ -770,6 +827,46 @@ export class Toolbar extends BaseModuleInstance implements ModuleInstance {
         this.ctx.toolbarManager.pressButton(group, this.buttonId({ id, value }), value, this.buttonRect(button), event);
     }
 
+    private onDragStart(event: MouseEvent, group: ToolbarGroup) {
+        const element = this.elements[ToolbarPosition.Floating];
+
+        event.preventDefault();
+        event.stopPropagation();
+
+        this.dragState = {
+            client: Vec2.from(event.clientX, event.clientY),
+            position: Vec2.from(
+                Number(element.style.getPropertyValue('left').replace('px', '')),
+                Number(element.style.getPropertyValue('top').replace('px', ''))
+            ),
+            detached: true,
+        };
+
+        const onDrag = (e: MouseEvent) => this.onDrag(e, group);
+        const window = getWindow();
+        window.addEventListener('mousemove', onDrag);
+        window.addEventListener('mouseup', () => window.removeEventListener('mousemove', onDrag), {
```

Comment:
I was having some issues with this with the dialog since some events are not propagated up beyond our container element. I suggest using the `domManager` instead of `window` as that contains all we care about and will always receive the events.

```suggestion
        const { ctx: { domManager } } = this;
        domManager.addEventListener('mousemove', onDrag);
        domManager.addEventListener('mouseup', () => domManager.removeEventListener('mousemove', onDrag), {
```

Reply:
Though, may not be needed now - https://github.com/ag-grid/ag-charts/pull/2347/commits/c17cc10d94298fc5b1e71c20dbd96fb5319f7017

Reply:
The issue I found with using domManager is that dragging will continue if you release the mouse (mouseup) outside of the canvas container, I did window so it behaves like the navigator drag